src/components/ErisPlayerManager.ts

```diff
@@ -0,0 +1,155 @@
+
+export class ErisPlayerManager extends Collection<Player> {
+  nodes: Node[]
+  #pending: Record<string, any>
+  regions: Record<string, string[]>
+  defaultRegion: string
+  constructor (nodes: NodeConstructor[], options: PlayerManagerOptions) {
+    super(Player)
+    this.nodes = nodes.map(x => {
+      return new Node({
+        shards: options.shards,
+        user: options.userId,
+        ...x
+      })
+    })
+    this.nodes.forEach(x => x.on('message', this.onNodeMessage.bind(this)))
+    this.#pending = {}
+    this.regions = options.regions ?? {
+      asia: ['singapore', 'hongkong', 'russia', 'japan', 'india', 'dubai'],
+      eu: ['europe', 'amsterdam', 'london', 'frankfurt', 'eu-central', 'eu-west', 'vip-amsterdam'],
+      us: ['us-west', 'us-east', 'us-central', 'us-south', 'brazil', 'vip-us-west', 'vip-us-east'],
+      africa: ['southafrica'],
+      australia: ['sidney']
+    }
+    this.defaultRegion = options.defaultRegion ?? 'global'
+  }
+
+  async join (guildID: Snowflake, channelID: Snowflake) {
+    const player = this.get(guildID)
+    if (player && player.connected) {
+      player.switchChannel(channelID)
+      return Promise.resolve(player)
+    }
+    return new Promise((resolve, reject) => {
+      this.#pending[guildID] = {
+        channelID: channelID,
+        res: resolve,
+        rej: reject,
+        timeout: setTimeout(() => {
+          delete this.#pending[guildID]
+          reject(new Error('Voice connection timeout'))
+        }, 10000)
+      }
+    })
+  }
+
+  leave (guildID: Snowflake) {
+    const player = this.get(guildID)
+    if (!player) return
+    player.disconnect()
+    this.delete(guildID)
+  }
+
+  switch (guildID: Snowflake, channelID: Snowflake) {
+    const player = this.get(guildID)
+    if (!player) return
+    player.switchChannel(channelID)
+  }
+
+  voiceServerUpdate (data: any) {
+    if (this.#pending[data.guild_id] && this.#pending[data.guild_id].timeout) {
+      clearTimeout(this.#pending[data.guild_id].timeout)
+      this.#pending[data.guild_id].timeout = null
+    }
+    let player = this.get(data.guild_id)
+    if (!player) {
+      if (!this.#pending[data.guild_id]) {
+        return
+      }
+      player = new Player(this.selectBestLavalinkNode(data.endpoint), data.guild_id, (op, ctx) => data.shard.sendWS(op, ctx))
+      this.set(data.guild_id, player)
+    }
+    player.once('ready', () => {
+      if (this.#pending[data.guild_id]) {
+        this.#pending[data.guild_id].res(player)
+        delete this.#pending[data.guild_id]
+      }
+    })
+    player.once('disconnected', ctx => {
+      if (this.#pending[data.guild_id]) {
+        this.#pending[data.guild_id].rej(new Error(ctx.reason || 'Disconnected'))
+        delete this.#pending[data.guild_id]
+      }
+      this.delete(data.guild_id)
+    })
+    player.connect({
+      sessionId: data.session_id,
+      event: data
+    })
+  }
+
+  private selectBestLavalinkNode (endpoint: string) {
```

Comment:
Might want to document what "best" entails here, meaning how the algorithm works. From the looks of it, this seems to be by lowest system load in the current region.